bfv/encryptor.go

```diff
@@ -293,5 +304,38 @@ func (encryptor *skEncryptor) encrypt(plaintext *Plaintext, ciphertext *Cipherte
 	encryptor.gaussianSamplerQ.ReadAndAdd(ciphertext.value[0])
 
 	// ct = [-a*s + m + e , a]
-	ringQ.Add(ciphertext.value[0], plaintext.value, ciphertext.value[0])
+	if !plaintext.inZQ {
+		encryptor.TtoQ(plaintext)
+		encryptor.ringQ.Add(ciphertext.value[0], encryptor.polypool[0], ciphertext.value[0])
+	} else {
+		encryptor.ringQ.Add(ciphertext.value[0], plaintext.value, ciphertext.value[0])
+	}
+}
+
+func (encryptor *encryptor) TtoQ(p *Plaintext) {
```

Comment:
Should this method check that `!p.inZQ` ?

Reply:
I made those method private, as the check is done before.